drivers/ad5592r/ad5592r-base.c

```diff
@@ -0,0 +1,291 @@
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *  - Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *  - Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ *  - Neither the name of Analog Devices, Inc. nor the names of its
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+#include "ad5592r-base.h"
+#include "platform_drivers.h"
+
+/**
+ * Get GPIO value
+ *
+ * @param dev - The device structure.
+ * @param offset - The channel number.
+ * @return 0 or 1 depending on the GPIO value.
+ */
+int32_t ad5592r_gpio_get(struct ad5592r_dev *dev, uint8_t offset)
+{
+	int32_t ret = 0;
+	uint8_t val;
+
+	if (!dev)
+		return FAILURE;
+
+	if (dev->gpio_out & BIT(offset))
+		val = dev->gpio_val;
+	else
+		ret = dev->ops->gpio_read(dev, &val);
+
+	if (ret < 0)
+		return ret;
+
+	return !!(val & BIT(offset));
+}
+
+/**
+ * Set GPIO value
+ *
+ * @param dev - The device structure.
+ * @param offset - The channel number.
+ * @param value - the GPIO value (0 or 1)
+ */
+void ad5592r_gpio_set(struct ad5592r_dev *dev, uint8_t offset, int32_t value)
+{
+	if (!dev)
+		return FAILURE;
+
+	if (value)
+		dev->gpio_val |= BIT(offset);
+	else
+		dev->gpio_val &= ~BIT(offset);
+
+	dev->ops->reg_write(dev, AD5592R_REG_GPIO_SET, dev->gpio_val);
```

Comment:
why not return int ?
`return dev->ops->reg_write(dev, AD5592R_REG_GPIO_SET, dev->gpio_val);`
?